src/gui/MainWindow.cpp

```diff
@@ -464,6 +464,11 @@ void MainWindow::setMenuActionState(DatabaseWidget::Mode mode)
             m_ui->actionDatabaseMerge->setEnabled(m_ui->tabWidget->currentIndex() != -1);
 
             m_searchWidgetAction->setEnabled(true);
+
+            //if not already in search mode focus the search widget and select all its content
+            if (!inSearch) {
+                m_ui->toolBar->findChild<SearchWidget *>("SearchWidget")->searchFocus();
```

Comment:
The database widget signals can be linked in this function: https://github.com/keepassxreboot/keepassxc/blob/cd46a08f4a6b983ac7ee4052aadb6c2e9fe9713f/src/gui/SearchWidget.cpp#L117